### DISCUSSION TITLE
Status bar

### DISCUSSION THREAD

**Discussion by kybos** [muse-sequencer/muse#1163] [Team Posts]
I think MusE would deserve a status bar. There are several things that should or could go there, especially the CPU/DSP/XRUN "toolbar", which in fact is a global status display, not a tool. There are already so many other toolbars and buttons... XRUN reset could be done e.g. by double-click, no need for another button. 
And there would be room for message output on the left. In some cases better than just stderr/stdout where most users don't see them...
What do you think?

Something like that:
![image](https://user-images.githubusercontent.com/10009541/77093840-e48f4300-6a0b-11ea-849b-0dbb11d02dc1.png)

**Comment by spamatica** [comment]
I'm not against adding stuff, at least if the old way can be conserved. :)
However, I'm a bit manic about conserving screen estate so I would probably
disable it... even though I can see there is some info that would be good
to present there.

If I could I would remove toolbars until I had only one row (and now you
made me think about how to achieve that :D - I would want an alternative
version of the 'edit tools' toolbar as SNAP is the only feature on that one
that I use, then I could have only one row...)

Another thing that I sometimes wonder if we should find an alternative is
the Transport panel. A remnant from Cubase of 25-30 years ago... I don't
know, maybe it is an acquired taste and you can learn to use it, but I'm
thinking there is a reason it's not very common these days... personally I
never use it.

Den tors 19 mars 2020 kl 18:56 skrev kybos <notifications@github.com>:

**Comment by kybos** [comment]
> I'm not against adding stuff, at least if the old way can be conserved. :)

Probably not a good idea to have the CPU/XRUN in 2 different places/widgets...

> If I could I would remove toolbars until I had only one row (and now you
> made me think about how to achieve that :D - I would want an alternative
> version of the 'edit tools' toolbar as SNAP is the only feature on that one
> that I use, then I could have only one row...)

Yes, could be a good idea to separate the Snap dropdown.
Even if I am not so manic about screen space, I think there is a lot of room wasted by superfluous labels in the toolbars, where it is quite obvious what the widgets mean (Tempo, signature, cursor...). I removed some of them in my "experimental" branch:

![image](https://user-images.githubusercontent.com/10009541/77180669-8ecda000-6aca-11ea-9872-294133867f24.png)

> Another thing that I sometimes wonder if we should find an alternative is
> the Transport panel. A remnant from Cubase of 25-30 years ago... I don't
> know, maybe it is an acquired taste and you can learn to use it, but I'm
> thinking there is a reason it's not very common these days... personally I
> never use it.

Absolutely, this is a relic from times where there was no multitasking and screen resolutions of 640x480 (or even x400). I don't use it either because it's just too annoying, staying on top of other apps and can't be closed... The problem is that it contains functions that don't exist anywhere else, so there probably should be another toolbar for them, at least as an alternative.

There are many other "relics" and UI problems, I started trying to improve/change some of them in a separate branch, but after reading lot of the older issues and topics from the mailing list, I got the impression that the MusE team is more on the conservative side in this respect so it would be difficult to find a consensus ;-).

**Comment by spamatica** [comment]
Den fre 20 mars 2020 kl 17:30 skrev kybos <notifications@github.com>:

> I'm not against adding stuff, at least if the old way can be conserved. :)
>
> Probably not a good idea to have the CPU/XRUN in 2 different
> places/widgets...
>
Do you think it would consume resources, cpu?
That would be an argument against but otherwise I don't see why we can't
have alternative approaches. Same way we have toolbar, menu alternative and
keyboard shortcuts. People have different workflows.
We could move to a status-bar design but keep a toolbar around that is not
enabled by default (for people like me that may not want the statusbar)

I could go further and say that we might even have different versions of
toolbars, like the toolbar in question could exist as 'Edit Tools' and
'Edit Tools Extended', don't know if that specific toolbar makes sense
besides to prove a point. My personal opinion is that toolbars (and other
kinds of presentation/functionality too) can overlap to some extent.
I can't remember what application it was but I remember back in the day
there was at least some developer application for Windows (Delphi?) that
had completely modular toolbars. You had all the toolbuttons and were free
to compose your own toolbars from scratch. A bit overkill but pretty cool.

If I could I would remove toolbars until I had only one row (and now you
> made me think about how to achieve that :D - I would want an alternative
> version of the 'edit tools' toolbar as SNAP is the only feature on that one
> that I use, then I could have only one row...)
>
> Yes, could be a good idea to separate the Snap dropdown.
>
Yes, that's another way, pretty reasonable.

> Even if I am not so manic about screen space, I think there is a lot of
> room wasted by superfluous labels in the toolbars, where it is quite
> obvious what the widgets mean (Tempo, signature, cursor...). I removed some
> of them in my "experimental" branch:
>
> [image: image]
> <https://user-images.githubusercontent.com/10009541/77180669-8ecda000-6aca-11ea-9872-294133867f24.png>
>
> Looks good!

> Another thing that I sometimes wonder if we should find an alternative is
> the Transport panel. A remnant from Cubase of 25-30 years ago... I don't
> know, maybe it is an acquired taste and you can learn to use it, but I'm
> thinking there is a reason it's not very common these days... personally I
> never use it.
>
> Absolutely, this is a relic from times where there was no multitasking and
> screen resolutions of 640x480 (or even x400). I don't use it either because
> it's just too annoying, staying on top of other apps and can't be closed...
> The problem is that it contains functions that don't exist anywhere else,
> so there probably should be another toolbar for them, at least as an
> alternative.
>
Yeah, haven't gone through them but that would probably be good.

> There are many other "relics" and UI problems, I started trying to
> improve/change some of them in a separate branch, but after reading lot of
> the older issues and topics from the mailing list, I got the impression
> that the MusE team is more on the conservative side in this respect so it
> would be difficult to find a consensus ;-).
>
Hehe, well..we're getting old and grumpy ;)
I'm not sure which changes you are thinking about specifically. The ones I
remember I think were more about personal chemistry than the actual
changes.. I could be wrong. Personally I really don't like redesign in the
way that many commercial programs do (I'm looking at you Photoshop!), feels
like they change things randomly to make it look new and to sell new
courses.

MusE is a bit long in the tooth but I don't think that is all bad, in the
same way that many big opensource applications are a bit ugly but
dependable (like Gimp, Audacity, VLC).
Still, we were offered an UI redesign some years ago by a graphics
designer, cant remember his name, and it was pretty neat! It would have
been a nice improvement. As I recall it fell mostly because it was such a
huge undertaking to implement, we simply did not have the resources.
If we can redesign/refresh a bit more incrementally it is hopefully more
doable... and we have a chance to acclimatize ;)

**Comment by terminator356** [comment]
Don't forget that the toolbars are configurable and can be hidden and moved
 to the sides and bottom as kybos has done with the timeline bar at the bottom in the picture.

>  back in the day there was at least some developer application for Windows (Delphi?) that
had completely modular toolbars.

I was a fanatical Delphi and C++Builder user. Its innovative design was the reason I 
 suggested to Florian that he include a setting like theirs in his GUI Style tab.
Don't forget he was the one who said "I think MusE wastes a lot of space" and
 proceeded to try and help the situation with that tab, and sharable toolbars and MDI windows.

> we were offered an UI redesign some years ago by a graphics designer, cant remember his name

That is Budislav the Artist.
See the original threads here: [https://github.com/muse-sequencer/muse/issues/557](url)
His website is here: [https://www.deviantart.com/budislavtvp](url)
His designs are heavily centered on dark backgrounds and a single light blue colour for some reason.
We got into some arguments over it after I said "Put some colour in there, this isn't the death star".
Eventually I incorporated some of his ideas - sliders instead of knobs (yep, thank him for that push),
 and a dash of blue colour. You'll note that the new Zyn Fusion UI is (I believe) entirely his design.
As you can see from most of his designs, including Zyn Fusion, light blue is his 'go to' colour.
If all the apps that he wanted to change followed his designs, we'd have a somewhat 'blue world'.

[Edit] I'll point out five words that made a difference here: "I am not a coder".
I felt at the time it was a bit like knocking on the door of a house someone has built and saying
 "Nice house but I think it should look like this".

It's a world of compromise. Of give and take. I try to please everyone but there's only
 so much I can do, or that we as a team can do, when we're trying to ensure solid
 working audio/midi engines and an extremely complex core with features that work
 in an ever changing landscape of plugins, Qt versions and bugs, C++ versions and obsolete code.

**Comment by kybos** [comment]
> We could move to a status-bar design but keep a toolbar around that is not
> enabled by default (for people like me that may not want the statusbar)

True. But the status bar is normally used for message output, infos, errors etc.. If you can hide it, then you would have to think of alternative output for these. All doable, sure, but perhaps not really necessary. And then, the status bar does not look especially nice on many themes, it takes more space than I would like it to, so I am not quite convinced myself...

> You had all the toolbuttons and were free
> to compose your own toolbars from scratch. A bit overkill but pretty cool.

Yes, several big IDEs and other apps have that. As you say, probably overkill, and hardly feasible in MusE IMO, with all these different GUI Styles and complicated toolbar handling... 

> [image: image]
>  <https://user-images.githubusercontent.com/10009541/77180669-8ecda000-6aca-11ea-9872-294133867f24.png>
> 
>  Looks good!

So we do it in master? I.e. remove labels from toolbars where the meaning is clear from the context (or from reading the tooltip once at last ;-) )?

> MusE is a bit long in the tooth but I don't think that is all bad, in the
> same way that many big opensource applications are a bit ugly but
> dependable (like Gimp, Audacity, VLC).

True. But many of the big OS applications are transforming (often because of the GTK2 to GTK3 move nightmare) and put a lot of effort into a clean and modern UI in the course (look at Inkscape beta, not to mention Blender). 
Don't get me wrong, I don't say it's all bad, just the opposite. I am convinced that MusE is the best midi sequencer for Linux, has a very good overall workflow and a lot of potential. Otherwise I would not spend so much time trying to improve it, obviously ;-).
On the other side, I would not underestimate the fact that the attitude toward UIs has changed radically in the last couple of years, mainly due to the spread of the mobile devices with their simple and clean UIs. The people (especially the younger ones) now expect exactly that also from desktop apps. Even companies that never cared (e.g. business software) are now in UX fever because of that (I have to know, I work for one of them ;-) ).
And my personal experience (my son and other young people): MusE has good functionality, but the UI looks like it was 20 years old. The other big point: Current version not easily installable, no PPA, AppImage...

> Still, we were offered an UI redesign some years ago by a graphics
> designer, cant remember his name, and it was pretty neat! It would have
> been a nice improvement.

It was Budislav. I read the discussions and the guy really knows what he is talking about, I could subscribe to every word he wrote. Have you still his designs by any chance? They are not available any more in SF, except for the mixer strip (also very nice). He seems involved in Zyn, Inkscape, LMMS...

**Comment by spamatica** [comment]
Den fre 20 mars 2020 kl 22:36 skrev kybos <notifications@github.com>:

> We could move to a status-bar design but keep a toolbar around that is not
> enabled by default (for people like me that may not want the statusbar)
>
> True. But the status bar is normally used for message output, infos,
> errors etc.. If you can hide it, then you would have to think of
> alternative output for these. All doable, sure, but perhaps not really
> necessary. And then, the status bar does not look especially nice on many
> themes, it takes more space than I would like it to, so I am not quite
> convinced myself...
>
Okay, maybe it's better to improve the toolbars then.
Another criticism of the status bar is that you can only present one line
of status so if there are several

A related idea of more practical nature is to get around to making our own
logger class so we can have logfiles (and maybe make it possible to display
the log from inside MusE).

>
> https://user-images.githubusercontent.com/10009541/77180669-8ecda000-6aca-11ea-9872-294133867f24.png
>
> Looks good!
>
> So we do it in master? I.e. remove labels from toolbars where the meaning
> is clear from the context (or from reading the tooltip once at last ;-) )?
>
Yes, I agree with this. :)


> MusE is a bit long in the tooth but I don't think that is all bad, in the
> same way that many big opensource applications are a bit ugly but
> dependable (like Gimp, Audacity, VLC).
>
> True. But many of the big OS applications are transforming (often because
> of the GTK2 to GTK3 move nightmare) and put a lot of effort into a clean
> and modern UI in the course (look at Inkscape beta, not to mention Blender).
>
Could be. Though I guess part of my point is that 'modern' in this context
means that it is updated to what is in fashion right now. In ten, or even
five, years, it won't be.

Maybe I have already talked about it, seen the SerenityOS videos on
youtube?
One guy writing an operating system from scratch with the gui aesthetics
from Windows 95 (or there around). Listening to him talking about this (or
anything he talks about really) is very interesting. It is not that Windows
95 is that "good looking" but that it is very well thought out, everything
has a purpose. ...and in that way sort of timeless...
Anyway, the videos are very nice and I have, not the least, learned a bunch
of new tricks with modern C++ :)

Don't get me wrong, I don't say it's all bad, just the opposite. I am
> convinced that MusE is the best midi sequencer for Linux, has a very good
> overall workflow and a lot of potential. Otherwise I would not spend so
> much time trying to improve it, obviously ;-).
>
Good of you to say that ^.^ ...even though there are always things to
improve, more than we can ever complete...


> On the other side, I would not underestimate the fact that the attitude
> toward UIs has changed radically in the last couple of years, mainly due to
> the spread of the mobile devices with their simple and clean UIs. The
> people (especially the younger ones) now expect exactly that also from
> desktop apps. Even companies that never cared (e.g. business software) are
> now in UX fever because of that (I have to know, I work for one of them ;-)
> ).
> And my personal experience (my son and other young people): MusE has good
> functionality, but the UI looks like it was 20 years old. The other big
> point: Current version not easily installable, no PPA, AppImage...
>
As my argument goes above, I fear 'modern' will change more quickly than we
can adapt..
But of course I understand, if it does not look attractive, it won't be the
first application people try...

As for ease of installation, yes, we need to get our act together.


> Still, we were offered an UI redesign some years ago by a graphics
> designer, cant remember his name, and it was pretty neat! It would have
> been a nice improvement.
>
> It was Budislav. I read the discussions and the guy really knows what he
> is talking about, I could subscribe to every word he wrote. Have you still
> his designs by any chance? They are not available any more in SF, except
> for the mixer strip (also very nice). He seems involved in Zyn, Inkscape,
> LMMS...
>
Looking through old mails I have found some!

This one was a (early I think) mockup for a one-window UI
![muse_single_view_mockup](https://user-images.githubusercontent.com/4476983/77236805-32e64280-6bc2-11ea-837f-aa2f44b5f7ce.png)


Here's a more traditional version of the arranger
![muse_arranger_mockup](https://user-images.githubusercontent.com/4476983/77236808-3c6faa80-6bc2-11ea-8dee-de1bdacc0f6a.png)

Mixer
![muse_mixer_mockup](https://user-images.githubusercontent.com/4476983/77236809-409bc800-6bc2-11ea-9ac3-a7acf9a5e120.png)

And this one from 2017, I believe is the most recent one (the previous are
from 2015), which comes closest to what we already have (curiously I don't
remember this.., pretty nice)
![muse_mixer_mockup2](https://user-images.githubusercontent.com/4476983/77236811-498c9980-6bc2-11ea-86d8-10fe3731b3ba.png)

**Comment by kybos** [comment]
Unfortunately the pictures are not accessible, there are only placeholders.

**Comment by spamatica** [comment]
Oh, silly, replied via mail, maybe that doesn't work with images...
New try:
This one was a (early I think) mockup for a one-window UI
![muse_single_view_mockup](https://user-images.githubusercontent.com/4476983/77236869-cddf1c80-6bc2-11ea-95a5-2133465980a7.png)


Here's a more traditional version of the arranger
![muse_arranger_mockup](https://user-images.githubusercontent.com/4476983/77236874-da637500-6bc2-11ea-8ab0-dbed747a13a6.png)

Mixer
![muse_mixer_mockup](https://user-images.githubusercontent.com/4476983/77236877-e0f1ec80-6bc2-11ea-86f2-b6659cd48b9a.png)

And this one from 2017, I believe is the most recent one (the previous are from 2015), which comes closest to what we already have (curiously I don't remember this.., pretty nice)
![muse_mixer_mockup2](https://user-images.githubusercontent.com/4476983/77236879-e6e7cd80-6bc2-11ea-9180-8975f150b202.png)

**Comment by kybos** [comment]
Thanks for the pictures (now the links to them appeared even in your first post, they were not there yesterday for some reason...).

About "modern": It's my fault, I used this negatively connotated word, but this is complete misunderstanding. I am really the last person chasing for fashions. What I mean is ergonomics, clean design, symmetry, ease on the eye. UI/UX design is an autonomous science (a mix of art and psychology), and there are universal rules and best practices which are fashion-independent. Of course almost every company/software tries to create its unique look on top of that, but that's another matter, and can become an end it itself in the worst case (like with Photoshop...).

Good you mention Win95: That's a beautiful, ergonomic and successful design indeed (I've always used this "classic" look even in later Windows versions, with a single colour background - so much about me and "fashions" ;-) ). The best prove is that you can still find this basic design in most desktops now, in small variations (except Apple, of course). 
I will have a look at SerenityOS, sounds interesting.

I guess my theoretical talk must be confusing, better to become more concrete ;-). I will start another thread with UI issues which IMO are not good and should be improved.

Some more concrete points here, to the pictures above:

1. I don't think the design pattern in the first mock-up (typical IDE design) is necessary or even suitable for MusE. It could be done that way of course, but the huge effort necessary for that is probably better invested in other things. Possibly the existing MDI GUI style could be improved to allow for more "integrated" user experience (it's also a little buggy now). But the SDI style also has its benefits, especially if you have several monitors.

2. The other pictures (except 4): That's really beautiful design. It's not about the colours (which I understand Tim doesn't like) or Breeze etc., which you can call fashion-dependent to some extent. But it's the overall consistency, the symmetry which make it easy and pleasing to the eye, and that's very important for relaxed working (and also for the first impression on potential new users...).
I of course don't think it feasible to throw the current UI away and make a completely new one, but a similar design should be the target. I don't even think we have a real problem with the track list and the canvases, which could be probably improved a little quite easily. The main point is the mixer IMO, which is not surprising, as I understand from what I read that you guys are deliberately trying to put as much information and elements in as small a space (width) as possible, which is quite contradictory to a relaxed and clean UI ;-).

I don't want to be pessimistic, but the comparison of MusE with other "ugly" apps like Gimp or Audacity is not quite correct, because there are no real alternatives for them in Linux. But we have QTractor with nice clean UI (just look at the mixer ;-) ), which will probably stop to crash permanently at some point; we have LMMS, very active development there, adding the badly missing functionality, with a good theme it also has a nice UI; and perhaps Ardour will add a proper midi editor in future (don't like the UI much personally, but it's OK).
